<a href="https://colab.research.google.com/github/Abdelrhman129/my-_colab_notbook/blob/main/Data_Anlalysis_Phython.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Analysis with Python: Part 1 of 6

**Part_one(frist_video)**

In [1]:
2+3+9

14

In [2]:
99-73

26

In [3]:
23.54 * -1432

-33709.28

In [4]:
100/7

14.285714285714286

In [5]:
100//7

14

In [6]:
100 % 7

2

In [7]:
5 ** 3

125

In [8]:
((2+5)*(17-3))/(4**3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page: https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . You can use the empty cells below and add more cells if required.

[ ]
Solving multi-step problems using variables
Let's try solving the following word problem using Python:

A grocery store sells a bag of ice for $1.25, and makes 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided, and gradually convert the word problem into a mathematical expression which can be evaluated using Python.

Cost of ice bag ($) = 1.25

Profit margin = 20% = .2

Profit per bag ($) = profit margin * cost of ice bag = .2 * 1.25

No. of bags = 500

Total profit = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [9]:
500*(1.25*2)

1250.0

In [11]:
cost_of_ice_bag = 1.25

In [12]:
profit_margin = .2

In [13]:
number_of_bags = 500

In [14]:
profit_margin

0.2

In [17]:
profit_per_bag = cost_of_ice_bag * profit_margin

In [18]:
total_profit = number_of_bags * profit_per_bag

In [19]:
total_profit

125.0

In [20]:
print("The grocery store makes a total profit of $",total_profit)

The grocery store makes a total profit of $ 125.0


In [21]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Evaluating conditions using Python
Apart from arithmetic operations, Python also provides several oprations for comparing numbers & variables.

Operator	Description
==	Check if operands are equal
!=	Check if operands are not equal
>	Check if left operand is greater than right operand
<	Check if left operand is less than right operand
>=	Check if left operand is greater than or equal to right operand
<=	Check if left operand is less than or equal to right operand
The result of a comparision operation is either True or False (note the uppercase T and F). These are special keywords in Python. Let's try out some experiment with comparision operators.

In [31]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [23]:
# Equality check_True
my_favorite_number == 1

True

In [24]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [25]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [26]:
# Not equal check - False
a_neutral_number != 3

False

In [27]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [28]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >=10
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evalute to `True`. Otherwise it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


In [33]:
my_favorite_number

1

In [34]:
my_favorite_number >0 and my_favorite_number <=3

True

In [37]:
my_favorite_number <0 and my_favorite_number <=3

False

In [38]:
my_favorite_number >0 and my_favorite_number >=3

False

In [39]:
True and False

False

In [40]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evalute to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [41]:
a_neutral_number

3

In [42]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [45]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [46]:
my_favorite_number < 0 or True

True

In [47]:
False or False

False

In [48]:
not a_neutral_number ==3

False

In [49]:
not False

True

In [50]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parantheses ( and ) to indicate the order in which logical operators should be applied.

In [51]:
(2>3 and 4<=5) or not (my_favorite_number <0 and True)

True

In [52]:
not (True and 0 < 1) or (False and True)

False

In [53]:
not True and 0 < 1 or False and True

False

# A Quick Tour of Variables and Data Types in Python

![](https://i.imgur.com/6cg2E9Q.png)


This tutorial series is a beginner-friendly introduction to programming and data analysis using the Python programming language. These tutorials take a practical and coding-focused approach. The best way to learn the material is to execute the code and experiment with it yourself.